content/en/developers/local-network.md

```diff
@@ -14,22 +14,140 @@ aliases:
+
+### Linux
+
+#### System-specific software dependencies
+
+Building Lotus requires some system dependencies, usually provided by your distribution.
+
+Arch:
+
+```shell
+sudo pacman -Syu opencl-icd-loader gcc git bzr jq pkg-config opencl-icd-loader opencl-headers opencl-nvidia hwloc
+```
+
+Ubuntu/Debian:
+
+```shell
+sudo apt install mesa-opencl-icd ocl-icd-opencl-dev gcc git bzr jq pkg-config curl clang build-essential hwloc libhwloc-dev wget -y && sudo apt upgrade -y
+```
+
+Fedora:
+
+```shell
+sudo dnf -y install gcc make git bzr jq pkgconfig mesa-libOpenCL mesa-libOpenCL-devel opencl-headers ocl-icd ocl-icd-devel clang llvm wget hwloc hwloc-devel
+```
+
+OpenSUSE:
+
+```shell
+sudo zypper in gcc git jq make libOpenCL1 opencl-headers ocl-icd-devel clang llvm hwloc && sudo ln -s /usr/lib64/libOpenCL.so.1 /usr/lib64/libOpenCL.so
+```
+
+Amazon Linux 2:
+
+```shell
+sudo yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm; sudo yum install -y git gcc bzr jq pkgconfig clang llvm mesa-libGL-devel opencl-headers ocl-icd ocl-icd-devel hwloc-devel
+```
+
+#### Rustup
+
+Lotus needs [rustup](https://rustup.rs). The easiest way to install it is:
+
+```shell
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+#### Go
+
+To build Lotus, you need a working installation of [Go 1.17.9 or higher](https://golang.org/dl/):
+
+```shell
+wget -c https://golang.org/dl/go1.17.9.linux-amd64.tar.gz -O - | sudo tar -xz -C /usr/local
+```
+
+{{< alert icon="tip">}}
+You'll need to add `/usr/local/go/bin` to your path. For most Linux distributions you can run something like:
+
+```shell
+echo 'export PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc && source ~/.bashrc
+```
+
+See the [official Golang installation instructions](https://golang.org/doc/install) if you get stuck.
+{{< /alert >}}
+
+### MacOS
+
+#### Software dependencies
+
+You must have XCode and Homebrew installed to build Lotus from source. Lotus requires that X-Code CLI tools be installed before building the Lotus binaries.
+
+Check if you already have the XCode Command Line Tools installed via the CLI, run:
+
+```shell
+xcode-select -p
+```
+
+This should output something like:
+
+```plaintext
+/Library/Developer/CommandLineTools
+```
+
+If this command returns a path, then you have Xcode already installed! You can [move on to installing dependencies with Homebrew](#homebrew).
+
+   {{< alert icon="warning">}}
+   If the above command doesn't return a path, install Xcode:
+
+   ```shell
+   xcode-select --install
+   ```
+
+   {{< /alert >}}
+
+Next up is installing Lotus dependencies using Homebrew.
+
+#### Homebrew
+
+We recommend that macOS users use [Homebrew](https://brew.sh) to install each of the necessary packages.
+
+1. Use the command `brew install` to install the following packages:
+
+   ```shell
+   brew install go bzr jq pkg-config rustup hwloc coreutils
+   ```
+
+Next up is cloning the Lotus repository and building the executables.
+
```

Comment:
Maybe we should have "Build from source" on macOS instead, since Homebrew basically fails without tweaks for everyone on MacOS 12 or higher.